system/Helpers/url_helper.php

```diff
@@ -634,3 +634,51 @@ function mb_url_title(string $str, string $separator = '-', bool $lowercase = fa
 }
 
 //--------------------------------------------------------------------
+
+if (! function_exists('url_to'))
+{
+	/**
+	 * Get the full, absolute URL to a controller method
+	 * (with additional arguments)
+	 *
+	 * @param string $controller
+	 * @param mixed  ...$args
+	 *
+	 * @throws \CodeIgniter\Router\Exceptions\RouterException
+	 *
+	 * @return string
+	 */
+	function url_to(string $controller, ...$args): string
+	{
+		if (! $route = route_to($controller, ...$args))
+		{
+			if (strpos($controller, '::') !== false)
+			{
+				$class = explode('::', $controller)[0];
+
+				if (! class_exists($class))
+				{
+					$controller = service('routes')->getDefaultNamespace() . $controller;
+				}
+			}
+
+			if ($route = route_to($controller, ...$args))
+			{
+				return site_url($route);
+			}
+
+			$explode = explode('::', $controller);
+
+			if (isset($explode[1]))
+			{
+				throw new \CodeIgniter\Router\Exceptions\RouterException(lang('HTTP.controllerNotFound', [$explode[0], $explode[1]]));
+			}
+			else
```

Comment:
no need else as already throw early.

Reply:
Yes. Useless else here.

Reply:
@ImMaax It would be great if you could fix this one too, but if not then I will merge it tomorrow.

Reply:
@michalsn Removed the if/else and explode :)

Reply:
@ImMaax We have a little misunderstanding :) Everything was good except this "else", so:
```php
			if (isset($explode[1]))
			{
				throw new \CodeIgniter\Router\Exceptions\RouterException(lang('HTTP.controllerNotFound', [$explode[0], $explode[1]]));
			}

			throw new \CodeIgniter\Router\Exceptions\RouterException(lang('HTTP.invalidRoute', [$controller]));

```

Reply:
Ohh, haha, alright, will change this real quick